test/tests/functions/number.json

```diff
@@ -209,173 +209,6 @@
         }
       ]
     },
-    {
-      "src": ".match {$foo :number} one {{one}} * {{other}}",
```

Comment:
This file tests the function `:number` (and its friend `:integer`, which are specified as part of the standard function set*. Won't we need these tests for that purpose? The file is named appropriately for that.

Reply:
The tests removed from here rely on the category selection behaviour of `:number`, which is not strictly speaking required of an implementation, as we only [say](https://github.com/unicode-org/message-format-wg/blob/main/spec/registry.md#rule-selection) that "selection should be based on CLDR plural rule data". If that wiggle room is not intentional, then we should (separately from this PR) firm that up, and possibly introduce more specific tests for `:number` that check a wider range of locales.

These tests also mostly exercise the MF2 spec side of pattern selection, for which each case is now included in the `:test:select` tests.

Reply:
We probably should fix these tests to use CLDR data, if that's what we want. My primary concern is not to lose tests for something that **_is_** part of our collection of specifications, even if it isn't part of MF2's core spec.